test/jdk/sun/java2d/marlin/HugePolygonClipTest.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.BasicStroke;
+import java.awt.Color;
+import java.awt.Graphics2D;
+import java.awt.RenderingHints;
+import java.awt.geom.Path2D;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import javax.imageio.ImageIO;
+
+/**
+ * Clip rendering test
+ *
+ * @test
+ * @bug 8297230
+ * @summary verify that huge polygon is properly rasterized
+ */
+public class HugePolygonClipTest {
+
+    private static final double LARGE_X_COORDINATE = 4194304.250;
+    private static final int SCENE_WIDTH = 600;
+    private static final int SCENE_HEIGHT = 400;
+
+    private static final float WIDTH = 2.73f;
+
+    private static final int G_MASK = 0x0000ff00;
+    private static final int R_MASK = 0x00ff0000;
+    private static final int RGB_MASK = 0x00ffffff;
+
+    static final boolean SAVE_IMAGE = false;
+
+    public static void main(final String[] args) {
+
+        // First display which renderer is tested:
+        // JDK9 only:
+        System.setProperty("sun.java2d.renderer.verbose", "true");
+
+        // enable Marlin logging:
+        System.setProperty("sun.java2d.renderer.log", "true");
+        System.setProperty("sun.java2d.renderer.clip", "true");
+        System.setProperty("sun.java2d.renderer.subPixel_log2_X", "8");
+
+        System.out.println("HugePolygonClipTest: size = " + SCENE_WIDTH + " x " + SCENE_HEIGHT);
+
+        final BufferedImage image = new BufferedImage(SCENE_WIDTH, SCENE_HEIGHT, BufferedImage.TYPE_INT_ARGB);
```

Comment:
It looks fine. One additional question did we have any tests to validate how the "custom images" will work? For example the image similar to this [one](https://github.com/openjdk/jdk/blob/master/src/java.desktop/share/classes/java/awt/image/BufferedImage.java#L383) but using a different order of offsets, like: int[] bOffs = {1, 0, 2, 3}; For some other bug I have found that some our code does not expect "non-default" layouts.

Reply:
Thanks.
In this particular test I selected ARGB as a well-known format so it does not work with non-standard ARGB channels. I could have used getRed(), getGreen() values to use standard API to get each components.